### ISSUE TITLE
seg fault when running tests

### ISSUE THREAD

**Issue by lannelin** [alan-turing-institute/arc-model-similarities-phase-2#60]
[adding for documentation purposes in case of future errors for other users]

seg fault on Intel MacBook Pro. Poetry lock below.

<details><summary>click for poetry lock</summary>

generated with `poetry export -f requirements.txt --output requirements.txt`

<code>
adjusttext==0.8 ; python_version >= "3.10" and python_version < "4.0"
aiohttp==3.8.4 ; python_version >= "3.10" and python_version < "4.0"
aiosignal==1.3.1 ; python_version >= "3.10" and python_version < "4.0"
appdirs==1.4.4 ; python_version >= "3.10" and python_version < "4.0"
async-timeout==4.0.2 ; python_version >= "3.10" and python_version < "4.0"
attrdict==2.0.1 ; python_version >= "3.10" and python_version < "4.0"
attrs==23.1.0 ; python_version >= "3.10" and python_version < "4.0"
certifi==2023.5.7 ; python_version >= "3.10" and python_version < "4.0"
cfgv==3.3.1 ; python_version >= "3.10" and python_version < "4.0"
charset-normalizer==3.1.0 ; python_version >= "3.10" and python_version < "4.0"
click==8.1.3 ; python_version >= "3.10" and python_version < "4.0"
colorama==0.4.6 ; python_version >= "3.10" and python_version < "4.0" and sys_platform == "win32" or python_version >= "3.10" and python_version < "4.0" and platform_system == "Windows"
contourpy==1.1.0 ; python_version >= "3.10" and python_version < "4.0"
cycler==0.11.0 ; python_version >= "3.10" and python_version < "4.0"
distlib==0.3.6 ; python_version >= "3.10" and python_version < "4.0"
docker-pycreds==0.4.0 ; python_version >= "3.10" and python_version < "4.0"
eagerpy==0.30.0 ; python_version >= "3.10" and python_version < "4.0"
exceptiongroup==1.1.1 ; python_version >= "3.10" and python_version < "3.11"
filelock==3.12.2 ; python_version >= "3.10" and python_version < "4.0"
fire==0.5.0 ; python_version >= "3.10" and python_version < "4.0"
fonttools==4.40.0 ; python_version >= "3.10" and python_version < "4.0"
foolbox==3.3.3 ; python_version >= "3.10" and python_version < "4.0"
frozenlist==1.3.3 ; python_version >= "3.10" and python_version < "4.0"
fsspec[http]==2023.6.0 ; python_version >= "3.10" and python_version < "4.0"
geomloss==0.2.6 ; python_version >= "3.10" and python_version < "4.0"
gitdb==4.0.10 ; python_version >= "3.10" and python_version < "4.0"
gitpython==3.1.31 ; python_version >= "3.10" and python_version < "4.0"
identify==2.5.24 ; python_version >= "3.10" and python_version < "4.0"
idna==3.4 ; python_version >= "3.10" and python_version < "4.0"
iniconfig==2.0.0 ; python_version >= "3.10" and python_version < "4.0"
jinja2==3.1.2 ; python_version >= "3.10" and python_version < "4.0"
joblib==1.2.0 ; python_version >= "3.10" and python_version < "4.0"
kiwisolver==1.4.4 ; python_version >= "3.10" and python_version < "4.0"
lightning-bolts==0.6.0.post1 ; python_version >= "3.10" and python_version < "4.0"
lightning-utilities==0.3.0 ; python_version >= "3.10" and python_version < "4.0"
llvmlite==0.40.0 ; python_version >= "3.10" and python_version < "4.0"
markupsafe==2.1.3 ; python_version >= "3.10" and python_version < "4.0"
matplotlib==3.7.1 ; python_version >= "3.10" and python_version < "4.0"
multidict==6.0.4 ; python_version >= "3.10" and python_version < "4.0"
munkres==1.1.4 ; python_version >= "3.10" and python_version < "4.0"
nodeenv==1.8.0 ; python_version >= "3.10" and python_version < "4.0"
numba==0.57.0 ; python_version >= "3.10" and python_version < "4.0"
numpy==1.23.5 ; python_version < "4.0" and python_version >= "3.10"
nvidia-cublas-cu11==11.10.3.66 ; python_version >= "3.10" and python_version < "4.0" and platform_system == "Linux"
nvidia-cuda-nvrtc-cu11==11.7.99 ; python_version >= "3.10" and python_version < "4.0" and platform_system == "Linux"
nvidia-cuda-runtime-cu11==11.7.99 ; python_version >= "3.10" and python_version < "4.0" and platform_system == "Linux"
nvidia-cudnn-cu11==8.5.0.96 ; python_version >= "3.10" and python_version < "4.0" and platform_system == "Linux"
opentsne==1.0.0 ; python_version >= "3.10" and python_version < "4.0"
otdd @ git+https://github.com/alan-turing-institute/arc-otdd.git@v1.0.0 ; python_version >= "3.10" and python_version < "4.0"
packaging==23.1 ; python_version >= "3.10" and python_version < "4.0"
pandas==2.0.2 ; python_version >= "3.10" and python_version < "4.0"
pathtools==0.1.2 ; python_version >= "3.10" and python_version < "4.0"
pillow==9.5.0 ; python_version >= "3.10" and python_version < "4.0"
platformdirs==3.7.0 ; python_version >= "3.10" and python_version < "4.0"
pluggy==1.2.0 ; python_version >= "3.10" and python_version < "4.0"
pot==0.9.0 ; python_version >= "3.10" and python_version < "4.0"
pre-commit==2.21.0 ; python_version >= "3.10" and python_version < "4.0"
protobuf==4.23.3 ; python_version >= "3.10" and python_version < "4.0"
psutil==5.9.5 ; python_version >= "3.10" and python_version < "4.0"
pynndescent==0.5.10 ; python_version >= "3.10" and python_version < "4.0"
pyparsing==3.1.0 ; python_version >= "3.10" and python_version < "4.0"
pytest-check==2.1.5 ; python_version >= "3.10" and python_version < "4.0"
pytest==7.3.2 ; python_version >= "3.10" and python_version < "4.0"
python-dateutil==2.8.2 ; python_version >= "3.10" and python_version < "4.0"
pytorch-lightning==1.8.3.post1 ; python_version >= "3.10" and python_version < "4.0"
pytz==2023.3 ; python_version >= "3.10" and python_version < "4.0"
pyyaml==6.0 ; python_version >= "3.10" and python_version < "4.0"
requests==2.31.0 ; python_version >= "3.10" and python_version < "4.0"
scikit-learn==1.2.2 ; python_version >= "3.10" and python_version < "4.0"
scipy==1.9.3 ; python_version >= "3.10" and python_version < "4.0"
seaborn==0.12.2 ; python_version >= "3.10" and python_version < "4.0"
sentry-sdk==1.25.1 ; python_version >= "3.10" and python_version < "4.0"
setproctitle==1.3.2 ; python_version >= "3.10" and python_version < "4.0"
setuptools==68.0.0 ; python_version >= "3.10" and python_version < "4.0"
six==1.16.0 ; python_version >= "3.10" and python_version < "4.0"
smmap==5.0.0 ; python_version >= "3.10" and python_version < "4.0"
tensorboardx==2.6.1 ; python_version >= "3.10" and python_version < "4.0"
termcolor==2.3.0 ; python_version >= "3.10" and python_version < "4.0"
threadpoolctl==3.1.0 ; python_version >= "3.10" and python_version < "4.0"
tomli==2.0.1 ; python_version >= "3.10" and python_version < "3.11"
torch==1.13.1 ; python_version >= "3.10" and python_version < "4.0"
torchmetrics==0.11.4 ; python_version >= "3.10" and python_version < "4.0"
torchtext==0.14.1 ; python_version >= "3.10" and python_version < "4.0"
torchvision==0.14.1 ; python_version >= "3.10" and python_version < "4.0"
tqdm==4.65.0 ; python_version >= "3.10" and python_version < "4.0"
typing-extensions==4.6.3 ; python_version >= "3.10" and python_version < "4.0"
tzdata==2023.3 ; python_version >= "3.10" and python_version < "4.0"
umap-learn==0.5.3 ; python_version >= "3.10" and python_version < "4.0"
urllib3==2.0.3 ; python_version >= "3.10" and python_version < "4.0"
virtualenv==20.23.1 ; python_version >= "3.10" and python_version < "4.0"
wandb==0.13.11 ; python_version >= "3.10" and python_version < "4.0"
wheel==0.40.0 ; python_version >= "3.10" and python_version < "4.0" and platform_system == "Linux"
yarl==1.9.2 ; python_version >= "3.10" and python_version < "4.0"
</code>
</details>

**Comment by lannelin** [comment]
resolved with setting numba env var to disable JIT:
`export NUMBA_DISABLE_JIT=1`
